XMLHttpRequest/open-url-bogus.htm

```diff
@@ -15,8 +15,10 @@
           assert_throws("SyntaxError", function() { client.open("GET", url) })
         }, document.title + " (" + url + ")")
       }
-      url("http:")
+      url("//[")
       url("http://a a/")
+      url("http:////////////")
```

Comment:
I think this is okay. This basically ends up in the host state at the end and because the scheme is special and we reached EOF the buffer is empty and therefore we return failure. It's rather tricky though.